docs/rules/no-this-in-before-route-enter.md

```diff
@@ -0,0 +1,72 @@
+# vue/no-this-in-before-route-enter
+
+> This rule prevents usage this in the "beforeRouteEnter" because this is undefined there. https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards
+
+## Rule Details
+
+Bad:
```

Comment:
Use `<eslint-code-block>` like any other docs to build your docs for online demos. You can try it locally with the `npm run docs:watch` command.